src/PublicApiAnalyzers/Core/Analyzers/DeclarePublicApiAnalyzer.cs

```diff
@@ -147,83 +156,118 @@ static void RegisterImplActions(CompilationStartAnalysisContext compilationConte
             }
         }
 
-        private static ApiData ReadApiData(List<(string path, SourceText sourceText)> data, bool isShippedApi)
+        private static ApiData ReadApiData(string path, SourceText sourceText, bool isShippedApi)
         {
-            var apiBuilder = ImmutableArray.CreateBuilder<ApiLine>();
-            var removedBuilder = ImmutableArray.CreateBuilder<RemovedApiLine>();
-            var maxNullableRank = -1;
+            var apiBuilder = ArrayBuilder<ApiLine>.GetInstance();
+            var removedBuilder = ArrayBuilder<RemovedApiLine>.GetInstance();
+            var lastNullableLineNumber = -1;
 
-            foreach (var (path, sourceText) in data)
-            {
-                int rank = -1;
+            // current line we're on.  Note: we ignore whitespace lines when computin gthis.
+            var lineNumber = -1;
 
-                foreach (TextLine line in sourceText.Lines)
-                {
-                    string text = line.ToString();
-                    if (string.IsNullOrWhiteSpace(text))
-                    {
-                        continue;
-                    }
+            var additionalFileInfo = new AdditionalFileInfo(path, sourceText, isShippedApi);
 
-                    rank++;
+            foreach (var line in sourceText.Lines)
+            {
+                // Skip whitespace.
+                var text = line.ToString();
+                if (string.IsNullOrWhiteSpace(text))
+                    continue;
 
-                    if (text == NullableEnable)
-                    {
-                        maxNullableRank = Math.Max(rank, maxNullableRank);
-                        continue;
-                    }
+                lineNumber++;
```

Comment:
>lineNumber++; [](http://example.com/codeflow?start=16&length=13)

I assume it's intentional that this doesn't count ws only / empty lines, but it seems odd.

Reply:
yup yup. im preserving semantics here

Reply:
Maybe @sharwell can verify it's the intended behavior, it definitely doesn't match what the variable name indicates to me.

Reply:
my preference is to *not* change this.  this was the behavior from before.  If we change it, we risk breaking things for some customer who does have blank lines and then the nullable directive :)